File: clients/python/synchronousImageTransmission.py
# Make sure to have the add-on "ZMQ remote API" running in
# CoppeliaSim. Do not launch simulation, but run this script

import time

from coppeliasim_zmqremoteapi_client import RemoteAPIClient


print('Program started')

client = RemoteAPIClient()
sim = client.require('sim')

sim.loadScene(sim.getStringParam(sim.stringparam_scenedefaultdir) + '/messaging/synchronousImageTransmissionViaRemoteApi.ttt')

visionSensorHandle = sim.getObject('/VisionSensor')
passiveVisionSensorHandle = sim.getObject('/PassiveVisionSensor')

sim.setStepping(True)
sim.startSimulation()

startTime = sim.getSimulationTime()
while sim.getSimulationTime() - startTime < 15:
    img, res = sim.getVisionSensorImg(visionSensorHandle)
    sim.setVisionSensorImg(passiveVisionSensorHandle, img)
    sim.step()
sim.stopSimulation()

print('Program ended')

File: clients/matlab/synchronousImageTransmission.m
% Make sure to have the add-on "ZMQ remote API" running in
% CoppeliaSim and have following scene loaded:
%
% scenes/messaging/synchronousImageTransmissionViaRemoteApi.ttt
%
% Do not launch simulation, but run this script
%

fprintf('Program started\n');

client = RemoteAPIClient();
sim = client.require('sim');

visionSensorHandle = sim.getObject('/VisionSensor');
passiveVisionSensorHandle = sim.getObject('/PassiveVisionSensor');

sim.setStepping(true);
sim.startSimulation();

startTime = sim.getSimulationTime();
t = startTime;
while t - startTime < 5
    [img, resX, resY] = sim.getVisionSensorCharImage(visionSensorHandle);
    % display image in MATLAB:
    %imshow(flip(permute(reshape(img, 3, resY, resX), [3 2 1]), 1));
    sim.setVisionSensorCharImage(passiveVisionSensorHandle, img);
    sim.step();
    t = sim.getSimulationTime();
end
sim.stopSimulation();

fprintf('Program ended\n');
